javascript/javascript3/week2/promise-visual homework folder/main.js

```diff
@@ -0,0 +1,57 @@
+const redCircle = document.querySelector('ul.marks li:nth-child(1)');
+const blueCircle = document.querySelector('ul.marks li:nth-child(2)');
+const greenCircle = document.querySelector('ul.marks li:nth-child(3)');
+
+// translateOneByOne function 
+
+function translateOneByOne() {
+    moveElement(redCircle, {
+            x: 20,
+            y: 300
+        })
+        .then(() => {
+            console.log('bluecircle moved from random start to its target position');
+        })
+        .then(() => moveElement(blueCircle, {
+            x: 400,
+            y: 300
+        }))
+        .then(() => {
+            console.log('bluecircle moved from random start to its target position');
+        })
+        .then(() => moveElement(greenCircle, {
+            x: 400,
+            y: 20
+        }))
+        .then(() => {
+            console.log('greencircle moved from random start to its target position');
+        });
+}
+translateOneByOne();//Each circle moved one by one
```

Comment:
when you work with promises, you can return a promise to the next `.then` like so:
```
// translateOneByOne function 

function translateOneByOne() {
    moveElement(redCircle, { x: 20, y: 300 })
        .then(() => {
            console.log('bluecircle moved from random start to its target position');
            return moveElement(blueCircle, { x: 400, y: 300 })
        })
        .then(() => {
            console.log('bluecircle moved from random start to its target position');
            return moveElement(greenCircle, { x: 400, y: 20 })
        })
        .then(() => { console.log('greencircle moved from random start to its target position'); });
}
translateOneByOne();//Each circle moved one by one 
```